ArduCopter/mode_turtle.cpp

```diff
@@ -45,11 +48,13 @@ void ModeTurtle::arm_motors()
     hal.rcout->disable_channel_mask_updates();
     change_motor_direction(true);
 
-    // disable throttle and gps failsafe
+    // disable throttle and gcs failsafe
     g.failsafe_throttle.set(FS_THR_DISABLED);
     g.failsafe_gcs.set(FS_GCS_DISABLED);
     g.fs_ekf_action.set(0);
 
+    // ensure the arming library is aware of our meddling
+    AP::arming().AP_Arming::arm(AP_Arming::Method::TURTLE_MODE, false);
```

Comment:
I have concerns around what happens if this fails... or the associated disarm...

You might still be in the state you're trying to avoid if this fails?